@navikt/aksel/src/codemod/transforms/darkside/darkside.utils.ts

```diff
@@ -0,0 +1,168 @@
+import type {
+  API,
+  ASTPath,
+  FileInfo,
+  ImportDeclaration,
+  JSXElement,
+  JSXOpeningElement,
+} from "jscodeshift";
+
+/**
+ * Finds a component import, accounting for sub-components and aliases.
+ * Returns the local name of the component. If the component is not found, returns null.
+ */
+function findComponentImport(input: {
+  j: API["j"];
+  file: FileInfo;
+  name: string;
+  packageType?: "react" | "tokens";
+}) {
+  const { j, file, name: _name, packageType = "react" } = input;
+
+  /* Account for sub-components */
+  const name = _name.includes(".") ? _name.split(".")[0] : _name;
+
+  const root = j(file.source);
+
+  let foundName: string | null = null;
+
+  root.find(j.ImportDeclaration).forEach((path) => {
+    if (packageType === "react" && !isAkselReactImport(path)) {
+      return;
+    }
+
+    if (packageType === "tokens" && !isAkselTokensImport(path)) {
+      return;
+    }
+
+    path.node.specifiers.forEach((specifier) => {
+      if (
+        specifier.type === "ImportSpecifier" &&
+        specifier.imported.name === name
+      ) {
+        foundName = specifier.local
+          ? specifier.local.name
+          : specifier.imported.name;
+      }
```

Comment:
I don't know if we already discussed this, and I am not even sure i want it, but just in case we do --> should we add an option for default imports?

```suggestion
    // Check for named imports
    path.node.specifiers.forEach((specifier) => {
      if (
        specifier.type === "ImportSpecifier" &&
        specifier.imported.name === name
      ) {
        foundName = specifier.local
          ? specifier.local.name
          : specifier.imported.name;
      }

      // Check for default imports. i.e.:
      // import Box from "@navikt/ds-react/Box"
      if (
        specifier.type === "ImportDefaultSpecifier" &&
        typeof path.node.source.value === "string" &&
        path.node.source.value.endsWith(`/${name}`)
      ) {
        foundName = specifier.local.name;
      }
```

Reply:
Haven't really found any cases where this is used, but could make sense just in case. If we add this, we will need to handle the name differently since
```
import * as Box from "@navikt/ds-react/Box`
```

leads to components being used like 
```
<Box.Box>
```